annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java

```diff
@@ -0,0 +1,354 @@
+		
+		Annotation inputAnno = parseTag(requestBody);
+		
+		// create indexed tag
+		Annotation storedAnno = createTag(requestBody);
+		
+		assertTrue(BodyInternalTypes.isSemanticTagBody(storedAnno.getBody().getInternalType()));
+		//validate the reflection of input in output!
+		validateOutputAgainstInput(storedAnno, inputAnno);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_URI, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET, TOTAL_BY_ID_FOUND);
+		
+		// validate fields
+		assertTrue(storedAnno.getMotivation().equals(MotivationTypes.TAGGING.name().toLowerCase()));
+		assertEquals(storedAnno.getBody().getInternalType(), BodyInternalTypes.SEMANTIC_TAG.name());
+		TagBody tagBody = ((SemanticTagBody) storedAnno.getBody());
+		assertNotNull(tagBody.getHttpUri());
+		assertTrue(tagBody.getHttpUri().equals("http://www.geonames.org/2988507"));
+		assertNotNull(tagBody.getLanguage());
+		assertTrue(tagBody.getLanguage().equals("en"));
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/09102/_UEDIN_214"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+
+	@Test
+	public void createGeoTag() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+		
+		String requestBody = getJsonStringInput(TAG_GEOTAG);
+		
+		Annotation inputAnno = parseTag(requestBody);
+		
+		// create indexed tag
+		Annotation storedAnno = createTag(requestBody);
+		
+		assertTrue(BodyInternalTypes.isGeoTagBody(storedAnno.getBody().getInternalType()));
+		//validate the reflection of input in output!
+		validateOutputAgainstInput(storedAnno, inputAnno);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET, TOTAL_BY_ID_FOUND);
+		
+		// validate fields
+		assertTrue(storedAnno.getMotivation().equals(MotivationTypes.TAGGING.name().toLowerCase()));
+		assertEquals(storedAnno.getBody().getInternalType(), BodyInternalTypes.GEO_TAG.name());
+		EdmPlace placeBody = (EdmPlace) ((EdmPlaceBody) storedAnno.getBody()).getPlace();
+		assertNotNull(placeBody.getLatitude());
+		assertTrue(placeBody.getLatitude().equals("48.853415"));
+		assertNotNull(placeBody.getLongitude());
+		assertTrue(placeBody.getLongitude().equals("-102.348800"));
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/09102/_UEDIN_214"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+
+	@Test
+	public void createTag() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```

Comment:
this is a search not a create test, please rename

Reply:
changed to search